pulsar/producer_impl.go

```diff
@@ -177,7 +177,14 @@ func (p *producer) internalCreatePartitionsProducers() error {
 
 	if oldProducers != nil {
 		oldNumPartitions = len(oldProducers)
-		if oldNumPartitions == newNumPartitions {
+		if newNumPartitions < oldNumPartitions {
```

Comment:
I didn't think it was possible for the number of partitions to decrease. What the topic deleted or did the broker return the wrong number of partitions? Any logs on the broker side to indicate that? If this is ignored will the next check fix it?

Reply:
Thank you for reply.
- There is no forced topic deletion or other admin operation this day.
- [brokerDeleteInactiveTopicsEnabled](https://github.com/apache/pulsar/blob/master/conf/broker.conf#L156) has configured as false.
- proxy and broker logs were checked, and there is not much valuable information, most is INFO level messages.
- only 20% producers has this error, it's weird.

maybe there is a concurrency issue in broker ownership cache that leads to data corruption?

In fact, client no need to do this detection, but the server needs to locate and solve the decrease problem.